### PR TITLE
Sort search results

### PR DESCRIPTION
**Pull request by Rasbin**
PR for sort search results.

### CODE REVIEW

File: app/i18n/messages/sv.json [hunk 1]
```diff
@@ -259,6 +259,7 @@
   "ShowResourcesLink.text": "Visa alla utrymmen och apparater",
   "SideNavbar.close": "Stäng",
   "SideNavbar.toggle": "Gäst",
+  "SortBy.label": "sortera efter:",
```
Comment (midori0507) [inline app/i18n/messages/sv.json]: Typo. `s` should be in uppercase

File: app/pages/search/SearchPage.js [hunk 2]
```diff
@@ -96,6 +106,15 @@ class UnconnectedSearchPage extends Component {
             showMap={showMap}
           />
         )}
+
+        <div className="container">
```
Comment (midori0507) [inline app/pages/search/SearchPage.js]: You don't put container in random place like this, either you make the whole component as a big container, or not using it at all

File: app/pages/search/Sort.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+class Sort extends Component {
+  constructor(props) {
+    super();
+    this.state = {
+      selected: '',
+    };
+
+    const { lang } = props;
+    this.sortOptions = [
+      { label: 'Name', value: `resource_name_${lang}` },
```
Comment (midori0507) [inline app/pages/search/Sort.js]: Make value as constants
```
const SORT_BY = {
    TYPE: 'type',
    ...
}
```

File: app/pages/search/Sort.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      { label: 'Type', value: 'type' },
+      { label: 'Premises', value: `unit_name_${lang}` },
+      { label: 'People', value: 'people_capacity' },
+      { label: 'Open now', value: 'sortByOpenNow' },
+    ];
+  }
+
+  handleChange = ({ value }) => {
+    const filters = {};
+    if (value === 'sortByOpenNow') {
```
Comment (midori0507) [inline app/pages/search/Sort.js]: Use constant defined above for `sortByOpenNow `

File: app/pages/search/Sort.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      const now = (new Date()).toISOString();
+      filters.orderBy = null;
+      filters.available_between = `${now},${now}`;
```
Comment (midori0507) [inline app/pages/search/Sort.js]: {now} {now} ?

File: app/utils/searchUtils.js [hunk 6]
```diff
@@ -18,9 +18,12 @@ function getFetchParamsFromFilters(filters) {
       filters.duration
     ),
     { purpose: filters.purpose === 'all' ? '' : filters.purpose },
-    { page: filters.page || 1 }
+    { page: filters.page || 1 },
+    { order_by: filters.orderBy || '' }
```
Comment (midori0507) [inline app/utils/searchUtils.js]: should have 1 default filter active instead of ``

File: app/i18n/messages/sv.json (hunk 1 above)
Comment (Rasbin) [reply app/i18n/messages/sv.json]: Corrected in new commit

File: app/pages/search/Sort.js (hunk 4 above)
Comment (Rasbin) [reply app/pages/search/Sort.js]: Corrected in new commits

File: app/utils/searchUtils.js (hunk 6 above)
Comment (Rasbin) [reply app/utils/searchUtils.js]: Made simple refactor for this in new commit. Did not include order by filter at all if no orderBy value is present. I hope this is fine.

File: app/pages/search/Sort.js (hunk 3 above)
Comment (Rasbin) [reply app/pages/search/Sort.js]: Corrected in new commits

File: app/pages/search/Sort.js (hunk 5 above)
Comment (Rasbin) [reply app/pages/search/Sort.js]: There is this `available_between` in API documentation which accepts / requires 2 values. It is just comparing the current time to check the currently opened resource. So, these two `now` is  a trick to get the intended result of open now resource as `available_between accepts 2 values`.

Comment (midori0507) [reply app/pages/search/Sort.js]: I discussed with backend guy, make the `endTime` as the end of the day. Next time, don't do trick.

File: app/pages/search/SearchPage.js (hunk 2 above)
Comment (Rasbin) [reply app/pages/search/SearchPage.js]: Removed the div with the container and wrapped Sort and SearchResults with Page Wrapper in the recent commit.

File: app/pages/search/Sort.js (hunk 5 above)
Comment (Rasbin) [reply app/pages/search/Sort.js]: End of the day added in the new commit as suggested. But I am not receiving the data from API, if you could have a look.

Comment (midori0507) [reply app/pages/search/Sort.js]: @Rasbin: No idea, let me know only when PR is ready to review.

Comment (Rasbin) [reply app/pages/search/Sort.js]: Sort By open now removed in the recent commit due to no API support.

File: app/utils/searchUtils.js [hunk 7]
```diff
@@ -18,9 +18,10 @@ function getFetchParamsFromFilters(filters) {
       filters.duration
     ),
     { purpose: filters.purpose === 'all' ? '' : filters.purpose },
-    { page: filters.page || 1 }
+    { page: filters.page || 1 },
+    { ...(filters.orderBy ? { order_by: filters.orderBy } : {}) },
+    { ...(filters.available_between ? { available_between: filters.available_between } : {}) }
```
Comment (rBoost) [inline app/utils/searchUtils.js]: What is this `available_between` line for, is it leftovers from the `open_now` sort option?

Comment (Rasbin) [reply app/utils/searchUtils.js]: Yes, it was missed to remove. I have removed it in the recent commit.

File: app/actions/sortActions.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  buildAPIUrl,
+  getErrorTypeDescriptor,
+  getHeadersCreator,
+  getRequestTypeDescriptor,
+  getSuccessTypeDescriptor,
+} from 'utils/apiUtils';
+
+function sortResources(orderBy) {
+  return {
+    [CALL_API]: {
```
Comment (midori0507) [inline app/actions/sortActions.js]: Change CALL_API to RSAA, you're using `very old` API, remove your `node_modules` and do `yarn install` again @Rasbin

File: app/constants/AppConstants.js [hunk 9]
```diff
@@ -68,4 +68,10 @@ export default {
   TIME_FORMAT: 'H:mm',
   TIME_SLOT_DEFAULT_LENGTH: 30,
   TRACKING: SETTINGS.TRACKING,
+  SORT_BY_OPTIONS: {
+    RESOURCE_NAME: 'resource_name_lang',
+    TYPE: 'type_name_lang',
+    PREMISES: 'unit_name_lang',
+    PEOPLE: 'people_capacity',
```
Comment (midori0507) [inline app/constants/AppConstants.js]: put a TODO: add sortby `openingNow` feature comment for later development

File: app/pages/search/searchPageSelector.js [hunk 10]
```diff
@@ -21,6 +21,7 @@ const orderedSearchResultIdsSelector = createSelector(
   resourcesSelector,
   (resourceIds, resources) => {
     const selectedResources = resourceIds.map(id => resources[id]);
+    console.log(orderBy(selectedResources, 'distance').map(resource => resource.id));
```
Comment (midori0507) [inline app/pages/search/searchPageSelector.js]: Remove this console.log

File: app/state/reducers/ui/searchReducer.js [hunk 11]
```diff
@@ -41,6 +41,19 @@ function searchReducer(state = initialState, action) {
       });
     }
 
+    case types.API.SORT_RESULTS_GET_SUCCESS: {
+      const results = Object.keys(action.payload.entities.resources || {});
+      console.log(results, 'results');
```
Comment (midori0507) [inline app/state/reducers/ui/searchReducer.js]: Remove console.log

File: app/state/reducers/ui/searchReducer.js [hunk 12]
```diff
@@ -41,6 +41,19 @@ function searchReducer(state = initialState, action) {
...
+      const paginatedResources = Object.values(action.payload.entities.paginatedResources || {});
+      const resultCount = paginatedResources.length ? paginatedResources[0].count : 0;
+      return state.merge({
+        resultCount,
+        results,
+        searchDone: true,
+        unitId: null,
```
Comment (midori0507) [inline app/state/reducers/ui/searchReducer.js]: why unitId: `null` needed to be here?

File: app/utils/searchUtils.js [hunk 13]
```diff
@@ -18,9 +18,9 @@ function getFetchParamsFromFilters(filters) {
       filters.duration
     ),
     { purpose: filters.purpose === 'all' ? '' : filters.purpose },
-    { page: filters.page || 1 }
+    { page: filters.page || 1 },
+    { ...(filters.orderBy ? { order_by: filters.orderBy } : {}) },
```
Comment (midori0507) [inline app/utils/searchUtils.js]: this is bad, why just `filters.orderBy ? ... {order_by: filters.orderBy) : {}`?

File: app/pages/search/SearchPage.js [hunk 14]
```diff
@@ -96,7 +106,13 @@ class UnconnectedSearchPage extends Component {
             showMap={showMap}
           />
         )}
+
         <PageWrapper className="app-SearchPage__wrapper" title={t('SearchPage.title')} transparent>
+          <Row>
+            <Col className="app-SearchControlsContainer__control sortControl col-md-offset-8" md={4} sm={6}>
```
Comment (midori0507) [inline app/pages/search/SearchPage.js]: Dont use bootstrap like this, use offset in input props, not className
the `sortControl` need to follow naming pattern for example `app-SearchControl__sortControl`

File: app/constants/AppConstants.js (hunk 9 above)
Comment (Rasbin) [reply app/constants/AppConstants.js]: Done!

File: app/actions/sortActions.js (hunk 8 above)
Comment (Rasbin) [reply app/actions/sortActions.js]: Actually, found that this file re-appeared somehow which is not in use. So, I removed it in latest commit.

File: app/state/reducers/ui/searchReducer.js (hunk 12 above)
Comment (Rasbin) [reply app/state/reducers/ui/searchReducer.js]: This block is not used and has been removed.

File: app/utils/searchUtils.js (hunk 13 above)
Comment (Rasbin) [reply app/utils/searchUtils.js]: This line is now refactored in new commit. The idea of using the previous way was from here : http://2ality.com/2017/04/conditional-literal-entries.html

File: app/pages/search/SearchPage.js (hunk 14 above)
Comment (Rasbin) [reply app/pages/search/SearchPage.js]: Corrections made.

**Pull request merged**